### PR TITLE
Fix #1898: Indexing: periodic "Built Range" log message is broken

### LINKED ISSUE #1898: Indexing: periodic "Built Range" log message is broken

### PR DESCRIPTION
**Pull request by jjezra**

### CODE REVIEW

**Review by jleach4 — CHANGES_REQUESTED** [review]
Test?

**Comment by jjezra** [comment]
> Test?

I've verified that the log message appears in some of the existing tests. Example: 
```
2022-11-07 15:54:55,826 [INFO] c.a.f.r.p.f.IndexingBase - Indexer: Built Range calling_method="buildMultiTargetIndex" delay="1" indexer_id="c016639b-30cc-47d4-98bd-5810e2bd4186" indexing_method="multi target by records" limit="17" range_end="null" range_start="null" records_per_second="10000" records_scanned="17" subspace="\x15\x1a" target_index_name="[indexA, indexB, indexC, indexD]"
```

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java [hunk 1]
```diff
@@ -622,7 +624,7 @@ private void validateTimeLimit(int toWait) {
     private boolean shouldLogBuildProgress() {
         long interval = common.config.getProgressLogIntervalMillis();
         long now = System.currentTimeMillis();
-        if (interval == 0 || interval < (now - timeOfLastProgressLogMillis)) {
+        if (interval == 0 || interval > (now - timeOfLastProgressLogMillis)) {
```
[the remark this answers is not in the document]
Comment (jjezra) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java:627]: I can do that, but `TestHelpers.assertLogs` will not test the interval but just the existence of this message (appearing at first continuation - i.e. in any case of not indexing all in a single transaction).

**Comment by jjezra** [comment]
```FDBRecordStoreIndexTest > testBoundaryPrimaryKeys() FAILED```

Test this please

**Comment by jjezra** [comment]
Test this please

**Comment by jjezra** [comment]
Squashed.

**Review by alecgrieser — CHANGES_REQUESTED** [review]

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jjezra) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java:627]: ✅

File: fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerSimpleTest.java [hunk 2]
```diff
@@ -1051,4 +1052,41 @@ public void testTimeLimit() {
...
+            records.forEach(recordStore::saveRecord);
+            context.commit();
+        }
+
+        Index index = new Index("newIndex", field("num_value_2").ungrouped(), IndexTypes.SUM);
+        FDBRecordStoreTestBase.RecordMetaDataHook hook = metaDataBuilder -> metaDataBuilder.addIndex("MySimpleRecord", index);
+        openSimpleMetaData(hook);
+        try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                .setDatabase(fdb).setMetaData(metaData).setIndex(index).setSubspace(subspace)
+                .setProgressLogIntervalMillis(10)
```
[the remark this answers is not in the document]
Comment (jjezra) [reply fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerSimpleTest.java:1073]: ✅

**Review by alecgrieser — APPROVED** [review]

**Review by jleach4 — APPROVED** [review]

**Pull request merged**